mubench.pipeline/utils/data_filter.py

```diff
@@ -0,0 +1,19 @@
+from typing import List
+
+
+class DataFilter:
```

Comment:
I'm not entirely happy with how the filtering works... although I'm not sure how a better solution would look like... from the pipeline point of view, we should allow filtering tasks, but this makes things much more complex, since filtering tasks, by definition, return the same type that they consume. To get rid of the duplication in the project/version/misuse tasks, we may be able to implement a filtering wrapper task. What do you think?

Reply:
The actual implementation is still buggy. I've spent way too much time to try and make filtering work this way, and I feel like this should actually be handled by the respective tasks. This abstract implementation just doesn't cut it. We should probably discuss this in our meeting.

Reply:
Agreed. Currently the problems are not clear to me, so it's hard to suggest a solution. But if it doesn't work yet, we definitely need another solution ;)